container_pipeline/workers/build.py

```diff
@@ -0,0 +1,89 @@
+#!/usr/bin/env python
+import hashlib
+import json
+import logging
+import os
+
+from container_pipeline.utils import Build, get_job_name
+from container_pipeline.lib.log import load_logger
+from container_pipeline.lib.openshift import Openshift, OpenshiftError
+from container_pipeline.workers.base import BaseWorker
+
+
+class BuildWorker(BaseWorker):
+    """Build worker"""
+
+    def __init__(self, logger=None, sub=None, pub=None):
+        super(BuildWorker, self).__init__(logger, sub, pub)
+        self.openshift = Openshift(logger=self.logger)
+
+    def handle_job(self, job):
+        """Handle job"""
+        parent_build_running = False
+        parents = job.get('depends_on', '').split(',')
+        parents_in_build = []
+        for parent in parents:
+            is_build_running = Build(parent).is_running()
+            if is_build_running:
+                parents_in_build.append(parent)
+            parent_build_running = parent_build_running or \
+                is_build_running
+
+        if parent_build_running:
+            self.logger.info('Parents in build: %s, pushing job: %s back '
+                             'to queue' % (parents_in_build, job))
+            self.queue.put(json.dumps(job), 'master_tube')
+        else:
+            self.logger.info('Starting build for job: %s' % job)
+            success = self.build(job)
+            if success:
+                self.handle_build_success(job)
+            else:
+                self.handle_build_failure(job)
+
+    def build(self, job):
+        """Build job"""
+        namespace = get_job_name(job)
+        project = hashlib.sha224(namespace).hexdigest()
+
+        try:
+            self.openshift.login()
+            build_id = self.openshift.build(project, 'build')
+            if not build_id:
```

Comment:
When would we hit something like this where `if not build_id:` is satisfied? Just trying to understand. Because in this scenario, I don't see we are logging the reason for failure. Which in turn leaves admins and end user unaware of the cause of failure.